frontend/src/metabase/embedding-sdk/theme/dynamic-css-vars.ts

```diff
@@ -0,0 +1,86 @@
+import { css } from "@emotion/react";
+
+import { alpha, darken, isDark, isLight, lighten } from "metabase/lib/colors";
+import type { MantineTheme } from "metabase/ui";
+
+import type {
+  ColorOperation,
+  SourceColorKey,
+} from "../types/private/css-variables";
+
+import { DYNAMIC_CSS_VARIABLES } from "./dynamic-css-vars-config";
+
+const isColorDefined = (color: string) =>
+  color && color !== "transparent" && color !== "unset";
+
+/**
+ * Determine if the current color scheme is dark based on the palette.
+ */
+export function getIsDarkThemeFromPalette(theme: MantineTheme) {
+  const backgroundColor = theme.fn.themeColor("background");
+  const foregroundColor = theme.fn.themeColor("text-dark");
+
+  // Dark foreground color indicates a dark theme.
```

Comment:
```suggestion
  // Dark background color indicates a dark theme.
```

I was confused for a sec.

Reply:
Nice typo catch 🎣